Test failures on python2

When invoking 'python setup.py test' under Python 2.7.13 on a Fedora 25 machine, two tests are failing:

The issues do not appear when invoking the tests with python3.

FAIL: test_time_in_std_mods (ropetest.builtinstest.BuiltinModulesTest)
Traceback (most recent call last):
  File "/home/<snip>/fedora-scm/python-rope/rope-0.10.5/ropetest/builtinstest.py", line 503, in test_time_in_std_mods
    self.assertTrue('time' in rope.base.stdmods.standard_modules())
AssertionError: False is not true

FAIL: test_sorting_imports_and_standard_modules2 (ropetest.refactor.importutilstest.ImportUtilsTest)
Traceback (most recent call last):
  File "/home/<snip>/fedora-scm/python-rope/rope-0.10.5/ropetest/refactor/importutilstest.py", line 814, in test_sorting_imports_and_standard_modules2
    self.import_tools.sort_imports(pymod))
AssertionError: 'import sys\nimport time\n' != u'import sys\n\nimport time\n'

--- a/rope/base/stdmods.py
+++ b/rope/base/stdmods.py
@@ -43,6 +43,10 @@ def normalize_so_name(name):
     """
     if "cpython" in name:
         return os.path.splitext(os.path.splitext(name)[0])[0]
+    # XXX: Special handling for Fedora python2 distribution
+    # See: https://github.com/python-rope/rope/issues/211
+    if name == "timemodule.so":
+        return "time"
     return os.path.splitext(name)[0]
 
 

--- a/ropetest/builtinstest.py
+++ b/ropetest/builtinstest.py
@@ -502,6 +502,11 @@ class BuiltinModulesTest(unittest.TestCase):
         import rope.base.stdmods
         self.assertTrue('time' in rope.base.stdmods.standard_modules())
 
+    def test_timemodule_normalizes_to_time(self):
+        import rope.base.stdmods
+        self.assertEqual(
+            rope.base.stdmods.normalize_so_name('timemodule.so'), 'time')
+
 
 def suite():
     result = unittest.TestSuite()

